### PR TITLE
Object cache - create better debug

### PR DESCRIPTION
**Pull request by timotei-litespeed**
Improvements observed from issue https://github.com/litespeedtech/lscache_wp/issues/552

### CODE REVIEW

File: src/object-cache.cls.php [hunk 1]
```diff
@@ -133,6 +134,26 @@ public function __construct($cfg = false)
...
+	 */
+	private function debug_function( $text, $error = false ){
+	    if( defined('LSCWP_LOG') ){
+	        Debug2::debug( $text );
+	    }
+	    else{
+	        // Show messages if WP_DEBUG is on.
+	        $show = defined('WP_DEBUG') ? WP_DEBUG : false;
+	        // Show if text is error.
+	        $error && $show = true;
```
Comment (hi-hai) [inline src/object-cache.cls.php:152]: No reason to use this way, not good to read. 

Also, no need to use extra var, just reuse $error.

File: src/object-cache.cls.php [hunk 2]
```diff
@@ -358,15 +375,15 @@ private function _connect()
 
 		// If failed to connect
 		if ($failed) {
-			defined('LSCWP_LOG') && Debug2::debug('[Object] Failed to connect ' . $this->_oc_driver . ' server!');
+			$this->debug_function('[Object] ❌ Failed to connect ' . $this->_oc_driver . ' server!', true);
```
Comment (hi-hai) [inline src/object-cache.cls.php:378]: This shouldn't be changed to error_log, otherwise may flood client server

File: src/object-cache.cls.php [hunk 3]
```diff
@@ -133,6 +134,26 @@ public function __construct($cfg = false)
...
 		}
 	}
+	
+	/**
+	 * Add debug.
+	 *
+	 * @since  6.3
+	 * @access private
+	 */
+	private function debug_function( $text, $error = false ){
```
Comment (hi-hai) [inline src/object-cache.cls.php:144]: just use `debug()`

File: src/object-cache.cls.php (hunk 1 above)
Comment (timotei-litespeed) [reply src/object-cache.cls.php]: Changed

File: src/object-cache.cls.php (hunk 3 above)
Comment (timotei-litespeed) [reply src/object-cache.cls.php]: changed the name to shorter

File: src/object-cache.cls.php [hunk 4]
```diff
@@ -133,6 +140,22 @@ public function __construct($cfg = false)
...
+	 */
+	private function debug_oc( $text, $error = false ){
+	    if( defined('LSCWP_LOG') ){
+	        Debug2::debug( $text );
+	    }
+	    else{
+	        $show = $error ? true : (bool) $this->_cfg_debug;
+	        $show && error_log( gmdate('m/d/y H:i:s') . ' - ' . $text . PHP_EOL, 3, WP_CONTENT_DIR . '/obj_debug.log' );
+	    }
+	}
```
[the remark this answers is not in the document]
Comment (timotei-litespeed) [reply src/object-cache.cls.php:159]: added

File: src/object-cache.cls.php [hunk 5]
```diff
@@ -133,6 +140,23 @@ public function __construct($cfg = false)
...
+	 * @since  6.3
+	 * @access private
+	 */
+	private function debug_oc( $text, $show_error = false ){
+	    if( defined('LSCWP_LOG') ){
+	        Debug2::debug( $text );
+			return;
+	    }
+
+		if (!$show_error && $this->_cfg_debug!=1) return;
```
Comment (hi-hai) [inline src/object-cache.cls.php:156]: If debug option set to `Admin IP Only`, shouldn't it still be allowed to log?

File: src/object-cache.cls.php [hunk 6]
```diff
@@ -475,7 +488,7 @@ public function set($key, $data, $expire)
 			} catch (\RedisException $ex) {
 				$res = false;
 				$msg = sprintf(__('Redis encountered a fatal error: %s (code: %d)', 'litespeed-cache'), $ex->getMessage(), $ex->getCode());
-				Debug2::debug('[Object] ' . $msg);
+				Debug2::debug('' . $msg);
```
Comment (hi-hai) [inline src/object-cache.cls.php:491]: What is the purpose to write like this?

File: src/object-cache.cls.php (hunk 5 above)
Comment (timotei-litespeed) [reply src/object-cache.cls.php]: changed

File: src/object-cache.cls.php (hunk 6 above)
Comment (timotei-litespeed) [reply src/object-cache.cls.php]: Made changes

**Pull request merged**